switchy/apps/__init__.py

```diff
@@ -1,6 +1,90 @@
+# vim: tabstop=4 softtabstop=4 shiftwidth=4 textwidth=80 smarttab expandtab
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 '''
 Built-in applications
 '''
+
+# Registry of switchy apps
+_apps = {}
+
+def app(*args, **kwargs):
+    '''Decorator to register switchy application classes
+       You can call it as:
+
+       @app
+       class CoolAppController(object):
+           pass
+
+       This will register the class as a switchy app
+       The name of the app defaults to the class name
+       The help for the app defaults to the class doc string
+
+       But you can also provide alternative help and name
+       decorator arguments to register your app with:
+
+       @app('Just try it!', 'CoolName')
```

Comment:
Also my first thought, but I decided to optimize for the most common use case, which is adding help and letting the class name as the app name. I don't care strongly about it, we can change it if you want. Honestly I think most of the time we'll end up sticking to the defaults, but I added the decorator arguments first, only later I decided to use class name and __doc__ as defaults